src/python/pants/core/util_rules/environments.py

```diff
@@ -340,23 +331,23 @@ async def get_target_for_environment_name(
         ),
     )
     wrapped_target = await Get(
-        WrappedTarget,
+        WrappedTargetForBootstrappingOnly,
         WrappedTargetRequest(address, description_of_origin=_description_of_origin),
     )
-    tgt = wrapped_target.target
+    tgt = wrapped_target.val
     if not tgt.has_field(CompatiblePlatformsField) and not tgt.has_field(DockerImageField):
         raise ValueError(
             softwrap(
                 f"""
                 Expected to use the address to a `_local_environment` or `_docker_environment`
                 target in the option `[environments-preview].names`, but the name
                 `{env_name.val}` was set to the target {address.spec} with the target type
-                `{wrapped_target.target.alias}`.
+                `{tgt.alias}`.
                 """
             )
         )
     return EnvironmentTarget(tgt)
 
 
 def rules():
-    return collect_rules()
+    return (*collect_rules(), QueryRule(ChosenLocalEnvironmentName, [Platform]))
```

Comment:
Rather than taking a `Platform`, `determine_local_environment` should probably be updated to consume `Platform.create_for_localhost` directly.

Basically: `determine_local_environment` is not environment/platform dependent itself: it should _always_ produce an environment for localhost.

Reply:
Ohhh that makes sense. Good call